homeassistant/components/intellifire/binary_sensor.py

```diff
@@ -0,0 +1,108 @@
+"""Support for Intellifire Binary Sensors."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Any
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import IntellifireDataUpdateCoordinator
+from .const import DOMAIN
+
+POWER = "on_off"
+TIMER = "timer_on"
+HOT = "is_hot"
+THERMOSTAT = "thermostat_on"
+FAN = "fan_on"
+LIGHT = "light_on"
+PILOT = "pilot_light_on"
+
+
+@dataclass
+class IntellifireSensorEntityDescriptionMixin:
+    """Mixin for required keys."""
+
+    value_fn: Callable[[Any], int | str | float | None]
+
+
+@dataclass
+class IntellifireBinarySensorEntityDescription(
+    BinarySensorEntityDescription, IntellifireSensorEntityDescriptionMixin
+):
+    """Describes a binary sensor entity."""
+
+
+INTELLIFIRE_BINARY_SENSORS: tuple[IntellifireBinarySensorEntityDescription, ...] = (
+    IntellifireBinarySensorEntityDescription(
+        key=POWER,  # This is the sensor name
```

Comment:
I don't think you need to create a constant just for the key if it is not used elsewhere.